astropy/coordinates/representation.py

```diff
@@ -1563,7 +1566,13 @@ def __init__(self, lon, lat, distance, differentials=None, copy=True):
         super().__init__(lon, lat, distance, copy=copy,
                          differentials=differentials)
         if self._distance.unit.physical_type == 'length':
-            self._distance = self._distance.view(Distance)
+            try:
+                self._distance = Distance(self._distance, copy=False)
+            except ValueError:
```

Comment:
There may be other sources of `ValueError` here.  Easiest thing to do is just check that the message mentions the distance being >=0...?